### ISSUE TITLE
gradient w.r.t. input checking in testSpConv3d failed with cuda:0, k=2, s=2, d=1

### ISSUE THREAD

**Issue by masszhou** [traveller59/spconv#213]
I reimplemented a simplified version of spconv which derived from this repo. 
during my work I found out one test case failed during multiple trials.
after some investigation I can NOT confirm between torch.nn.Conv3D and spconv.SparseConv3d which is correct.
Thus this thread is only a remind information. 
thanks for the sharing of the implementation.

my test enviroment is pytoch 1.6
test case is 'testSpConv3d' in file traveller59/spconv/test/test_conv.py
with parameter set [device='cuda:0', shape=[19, 18, 17], bs=1, IC=64, IO=32, k=2, s=2, p=0, d=1]
failure asset is the gradient w.r.t. input
'self.assertAllClose(din_np, din_sparse_np, atol=1e-4)'
the strange thing is it will not fail every time in multiple trials, but frequently in e.g. once per ten trials.
at the same time, intermediate gradient and output are all correct as always.
